ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -85,21 +85,81 @@ type RayClusterReconciler struct {
 // +kubebuilder:rbac:groups="rbac.authorization.k8s.io",resources=rolebindings,verbs=get;list;watch;create;delete
 // Reconcile used to bridge the desired state with the current state
 func (r *RayClusterReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
-	_ = r.Log.WithValues("raycluster", request.NamespacedName)
-	log.Info("reconciling RayCluster", "cluster name", request.Name)
+	var err error
+
+	// Try to fetch the Event instance
+	event := &v1.Event{}
+	if err = r.Get(context.TODO(), request.NamespacedName, event); err == nil {
+		return r.eventReconcile(request, event)
+	}
```

Comment:
should we add some information for not find the event or get event error?

Reply:
if there is an error getting the v1.Event, it is possible it is a RayCluster object. If we put log info here, it will be too many information printed.

Reply:
Do we still need this `eventReconcile` if we have liveness probe?

Reply:
as we discussed, now we use both liveness probe and readiness probe. 

readiness probe is used to discover early failures and recover if possible.

liveness probe is used as the last resort to recover the cluster by restarting failed worker/head node